# 0 for opt mapping->wxy, 1 for loc mapping->sy.
# TODO change yaml name
#TODO remove not used parmater!!!!
  slam_mode_switch: 1
  if_loop_to_end: 1
#  if_use_

  # the path of map save
#  you must change it to right PATH or it will crush!!!!!!!!
  save_map_path: "/home/wxy/seu/reborn/work/resultnew/test/"
#  save_map_path: "/media/sy/R/dataSet/ZhongKeXingChi/newbag/park_loop1_out/"
#  save_map_path: "/home/lsy/map/"
  if_need_first_position: 0

  # LOAM feature threshold
  edgeThreshold: 1.0
  surfThreshold: 0.1
  edgeFeatureMinValidNum: 10
  surfFeatureMinValidNum: 100

  # voxel filter paprams
  DownSampleModeSwitch: 1                       # add by SY, 0 for voxel filter, 1 for uniformSample,
# 0:
  odometrySurfLeafSize: 0.4                     # default: 0.4 - outdoor, 0.2 - indoor
  mappingCornerLeafSize: 0.2                    # default: 0.2 - outdoor, 0.1 - indoor
  mappingSurfLeafSize: 0.2                      # default: 0.4 - outdoor, 0.2 - indoor
  surroundingKeyframeDensity: 2.0               # meters, downsample surrounding keyframe poses
# 1:
  odometrySurfRadiusSize_US: 0.2                     # default: 0.4 - outdoor, 0.2 - indoor
  mappingCornerRadiusSize_US: 0.1                    # default: 0.2 - outdoor, 0.1 - indoor
  mappingSurfRadiusSize_US: 0.2                      # default: 0.4 - outdoor, 0.2 - indoor
  surroundingKeyframeDensity_US: 1.0               # meters, downsample surrounding keyframe poses

  # robot motion constraint (in case you are using a 2D robot)
  z_tollerance: 2                            # meters
  rotation_tollerance: 5                     # radia

  # CPU Params
  mappingProcessInterval: 0.09                  # seconds, regulate mapping frequency

  # Surrounding map
  surroundingkeyframeAddingDistThreshold: 1.0  # meters, regulate keyframe adding threshold
  surroundingkeyframeAddingAngleThreshold: 10.0  # an, regulate keyframe adding threshold

  surroundingKeyframeSearchRadius: 50.0         # meters, within n meters scan-to-map optimization (when loop closure disabled)

  # Loop closure
  loopClosureEnableFlag: true
  loopClosureFrequency: 1.0                     # Hz, regulate loop closure constraint add frequency
  surroundingKeyframeSize: 30                  # submap size (when loop closure enabled)
  historyKeyframeSearchRadius: 10.0          # meters, key frame that is within n meters from current pose will be considerd for loop closure
  historyKeyframeSearchTimeDiff: 150.0          # seconds, key frame that is n seconds older will be considered for loop closure
  historyKeyframeSearchNum: 10                  # number of hostory key frames will be fused into a submap for loop closure
  historyKeyframeFitnessScore: 0.5           # icp threshold, the smaller the better alignment

  #LM opt
  maxIters: 20

  #GTSAM noise
  GNSS_noise_rpy: [ 1e-1, 1e-1, 1e-1 ]
  GNSS_noise_xyz: [ 1e-2, 1e-2, 1e-2 ]

  LidarOdom_noise_rpy: [1e-1, 1e-1, 1e-1 ]
  LidarOdom_noise_xyz: [1e-1, 1e-1, 1e-1 ]

  DR_noise_rpy: [ 1e-1, 1e-1, 1e-1 ]
  DR_noise_xyz: [ 1e-1, 1e-1, 8 ]

  Loop_noise_rpy: [ 1e-3, 1e-3, 1e-3 ]
  Loop_noise_xyz: [ 1e-3, 1e-3, 1e-3 ]

  GNSS_noise_roll: 1e-1
  GNSS_noise_pitch: 1e-1
  GNSS_noise_yaw: 1e-1
  GNSS_noise_x: 1e-1
  GNSS_noise_y: 1e-1
  GNSS_noise_z: 1e-1
  LidarOdom_noise_roll: 1e-2
  LidarOdom_noise_pitch: 1e-2
  LidarOdom_noise_yaw: 1e-2
  LidarOdom_noise_x: 1e-2
  LidarOdom_noise_y: 1e-2
  LidarOdom_noise_z: 1e-2
  DR_noise_roll: 1e-1
  DR_noise_pitch: 1e-1
  DR_noise_yaw: 1e-1
  DR_noise_x: 1e-4
  DR_noise_y: 1e-4
  DR_noise_z: 1e-4
  GNSSupdateDR: 1
  loop_noise_roll: 1e-1
  loop_noise_pitch: 1e-1
  loop_noise_yaw: 1e-1
  loop_noise_x: 1e-4
  loop_noise_y: 1e-4
  loop_noise_z: 1e-4

